### PR TITLE
Document that the spec uses RFC 2119

### LINKED ISSUE #1927: Document that the spec uses RFC 2119
**Link to problem area**: Yes

**Issue**
The spec uses RFC 2119, but doesn't document that. I would expect based on [Travis' statement](https://github.com/matrix-org/matrix-spec/issues/1782#issuecomment-2284286870) that fixing this would be a clarification and thus not require the MSC process.

https://tools.ietf.org/html/rfc2119

### PR DESCRIPTION
**Pull request by HarHarLinks**
Contributed with my community hat on.

I added it to the index because the index itself already starts using RFC 2119 language in section 3.

### Pull Request Checklist

Resolves #1927

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [x] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1928--matrix-spec-previews.netlify.app

### CODE REVIEW

**Comment by HarHarLinks** [comment]
could you clarify where I should put the newsfragment for a change to the index?

File: content/_index.md [hunk 1]
```diff
@@ -99,6 +99,22 @@ services - be that for instant messages, VoIP call setups, or any other
 objects that need to be reliably and persistently pushed from A to B in
 an interoperable and federated manner.
 
+### Bindingness of requirements
+
+As a standards document, the Matrix Spec needs to signify the bindingness of the
+individual features, APIs, procedures, or other requirements it describes.
```
Comment (richvdh) [inline content/_index.md:105]: "Bindingness" is rather an awkward word!

Can we just say something like:

```suggestion
### Use of RFC2119 keywords
```

... and omit the first paragraph, which I don't think adds much value?

File: content/_index.md [hunk 2]
```diff
@@ -99,6 +99,22 @@ services - be that for instant messages, VoIP call setups, or any other
...
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" across all parts of the
+specification are to be interpreted as described in
+[RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Some entire sections of the specification might be optional depending on the
+circumstances, for example the
```
Comment (richvdh) [inline content/_index.md:113]:
```suggestion
circumstances. For example, the
```

File: content/_index.md [hunk 3]
```diff
@@ -99,6 +99,22 @@ services - be that for instant messages, VoIP call setups, or any other
...
+[modules of the client-server API](/client-server-api/#modules) apply depending
+on the use case. The bindingness of the above key words appearing in such a
+section is only relevant if the section itself is applicable.
```
Comment (richvdh) [inline content/_index.md:116]:
```suggestion
on the use case. The requirement level expressed by the above key words appearing in such a
section is only relevant if the section itself is applicable.
```

**Review by richvdh — CHANGES_REQUESTED** [review]
some minor edits

File: content/_index.md (hunk 1 above)
Comment (HarHarLinks) [reply content/_index.md:105]: I tried to be more generic and may have ended up with a mangled Germenglish version. How do you think about something slightly more general and introductory than jumping straight to RFC 2119 in the title?

```suggestion
### Requirement levels
```

---

I'm slightly confused by the spelling in your proposal: Is it correct that contrary to MSCs, RFCs use a space before the number? RFC 2119 does where it references itself.

Comment (richvdh) [reply content/_index.md]: > I'm slightly confused by the spelling in your proposal: Is it correct that contrary to MSCs, RFCs use a space before the number?

No, that's just me being a crank. The space is correct.

**Review by richvdh — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**